Add route type and segments mode

chao-xian opened the pull request:
This is part of Content Tools team's 2 week blitz.

## Trello cards
* https://trello.com/c/XM9DuwVN
* https://trello.com/c/jmxMgLWf

Router data passes these fields to the publishing api. Short-URL-Manager will need to as well in order for us to support decommissioning of router-data.

Adds to the form for creating and updating redirects two new drop downs: `Route type` and `Segments mode`.

## Expected changes

## Before
![image](https://cloud.githubusercontent.com/assets/424772/25141413/66916210-245b-11e7-958b-01a24b29e38f.png)

## After
![image](https://cloud.githubusercontent.com/assets/424772/25141423/6c654828-245b-11e7-89c9-850cae9caa32.png)

cbaines commented:
What are your thoughts on the default value for segments mode?

If it is "preserve", for exact routes, the default behaviour would be to preserve the query parameters, and for prefix routes, it would be to preserve part of the path and query.

If it is "ignore", for prefix and exact routes would be to just redirect to the target url.

Based on my presumptions about how this is going to be used, "ignore" might make the best default?

pmanrubia commented:
@cbaines @chao-xian: This might be a good question for Ben H.

pmanrubia commented on app/controllers/short_url_requests_controller.rb:
```diff
@@ -80,10 +80,10 @@ def get_short_url_request
 
   def create_short_url_request_params
-    params[:short_url_request].permit(:from_path, :to_path, :reason, :organisation_slug, :confirmed)
+    params[:short_url_request].permit(:from_path, :to_path, :reason, :route_type, :organisation_slug, :confirmed)
```
What happens with existing redirects (in the local database) that do not have a route_type when the are listed or updated. Would we need a migration?

pmanrubia commented on app/models/redirect.rb:
```diff
@@ -10,6 +10,7 @@ class Redirect
   field :from_path, type: String
   field :to_path, type: String
   field :route_type, type: String
+  field :segments_mode, type: String
```
What happens with existing redirects (in the local database) that do not have a `segments_mode` when the are listed or updated. Would we need a migration?

pmanrubia commented:
@chao-xian It seems that my comments have been done on outdated code. As they are general questions I am not rewriting them.

chao-xian commented:
@cbaines I've addressed your comment in https://github.com/alphagov/short-url-manager/pull/78/commits/e5a9282c1d68f08673c12c82c61394c4eaee4964 (with Ben H's agreement)

chao-xian commented:
@pmanrubia have addressed in https://github.com/alphagov/short-url-manager/pull/78/commits/5d21b0b783b1ba71fd53193f4bc6d1c6cefa3de3
